src/transactions/TransactionFrame.cpp

```diff
@@ -735,25 +747,27 @@ TransactionFrame::insertKeysForTxApply(UnorderedSet<LedgerKey>& keys) const
 }
 
 void
-TransactionFrame::markResultFailed()
+TransactionFrame::markResultFailed(TransactionResult& txResult)
```

Comment:
Does this still need to be a member method of `TransactionFrame`? Seems like could be just a free internal function.

Reply:
This is also being called in `FuzzTransactionFrame::attemptApplication`.

Reply:
That's not a big issue, but I think in general it's better for classes to have less responsibilities when possible. Here it seems like the method no longer has anything to do with the `TransactionFrame` and could be either a public free function (given that it's reused), a static function or even removed altogether, given that it's one line and for some reason it's the only 'special' case for setting the result code (others seem to be just set in-place). In any case, that's up to your judgement.

Reply:
I do see your point.

Reply:
I am inclined to remove the method, since it is just one line of setting the result code, and we are already directly setting it for other kinds of code (other than `txFAILED`). The only thing I'm concerned is the comments in this function, which points out to some perf implication that caller might need to know. So I'm gonna keep this function for now and maybe think of a better place to put it later.